src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -0,0 +1,139 @@
+/*
+ *
+ * static void AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt);
+ *
+ */
+
+#include "postgres.h"
+
+#include "lib/stringinfo.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+#include "distributed/citus_ruleutils.h"
+
+static void AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt);
+const char * DeparseAlterRoleStmt(AlterRoleStmt *stmt);
+
+const char *
+DeparseAlterRoleStmt(AlterRoleStmt *stmt)
+{
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	AppendAlterRoleStmt(&buf, stmt);
+
+	return buf.data;
+}
+
+
+static void
+AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt)
+{
+	ListCell *optionCell = NULL;
+	RoleSpec *role = stmt->role;
+
+	appendStringInfo(buf, "ALTER ROLE ");
+	switch (role->roletype)
+	{
+		case ROLESPEC_CURRENT_USER:
+		{
+			appendStringInfo(buf, "CURRENT_USER");
```

Comment:
Well,  because Citus using a separate user while sending these commands to the remote nodes, writing `CURRENT_USER `  or `SESSION_USER ` wouldn't be correct. I think `RoleSpecString`  already resolves the names to actual role names. Can't we simply use that for all?

Reply:
I feel like this would not be the expected behaviour of a deparser. However, I still changed it and added a comment explaining this.

Reply:
Well, that's true. thanks for adding the comments.